### PR TITLE
Added scope parsing to TokenResponse

### PR DESCRIPTION
**Pull request by rojoiii**
Added scope parsing to fromResponseJson(...) as it is missing from parsing causing TokenResponse to lose scope values

---
This change is [Reviewable](https://reviewable.io/reviews/openid/appauth-android/131)

### CODE REVIEW

**Comment by rojoiii** [comment]
Addresses https://github.com/openid/AppAuth-Android/issues/130

File: library/javatests/net/openid/appauth/AuthorizationResponseTest.java [hunk 1]
```diff
@@ -133,5 +136,7 @@ private void checkExpectedFields(AuthorizationResponse authResponse) {
                 TEST_ID_TOKEN, authResponse.idToken);
         assertEquals("access token expiration time does not match",
                 TEST_TOKEN_EXPIRE_TIME, authResponse.accessTokenExpirationTime);
+        assertEquals("scope does not match",
+                     TEST_SCOPE, authResponse.scope);
```
Comment (iainmcgin) [inline library/javatests/net/openid/appauth/AuthorizationResponseTest.java]: You are testing the wrong class - your changes affect `TokenResponse`, not `AuthorizationResponse`. Please revert changes to `AuthorizationResponseTest` and add a test to `TokenResponseTest`.

**Comment by iainmcgin** [comment]
Can you, or your employer, sign the CLA? See [CONTRIBUTING.md](https://github.com/openid/AppAuth-Android/blob/master/CONTRIBUTING.md) for instructions.

**Comment by rojoiii** [comment]
I'll make the changes for the test, I already processed the CLA, maybe I did something wrong?

**Comment by iainmcgin** [comment]
I didn't see your name on the list of executors, there may be a delay. I'll check; meanwhile, please add the test to TokenResponseTest. Thanks for your help!

**Comment by rojoiii** [comment]
I'll go through the CLA process again just in case, I am not sure what case to add to TokenResponseTest as the fields/values/etc that are within it appeared to be tested in AuthorizationResponseTest which is why I had originally added scope testing there. If you could give me some direction on what I should add to TokenResponseTest I'll be happy to make the changes.

**Comment by iainmcgin** [comment]
You're adding parsing of the scope parameter to the token response, so you'll need to add the following tests:

1. That response JSON that does not include a 'scope' field parses to a TokenResponse object with null in `scope`.
2. That response JSON that includes a 'scope' field parses to a TokenResponse object with that value placed in `scope`.
3. Multiple scope values are handled correctly, and returned via `getScopeSet`.

You're right that TokenResponse requires a lot more tests; you can help us by adding these to get us moving :)

**Comment by rojoiii** [comment]
@iainmcgin I adjusted the scope parsing and added testing that I believe covers the cases you mentioned. Let me know if I need to make any further adjustments.

**Comment by iainmcgin** [comment]
Looks good now, thanks!

File: library/java/net/openid/appauth/TokenResponse.java [hunk 2]
```diff
@@ -216,6 +216,7 @@ public Builder fromResponseJson(@NonNull JSONObject json) throws JSONException {
                 }
                 setRefreshToken(JsonUtil.getStringIfDefined(json, KEY_REFRESH_TOKEN));
                 setIdToken(JsonUtil.getStringIfDefined(json, KEY_ID_TOKEN));
+                setScope(JsonUtil.getStringIfDefined(json, KEY_SCOPE));
```
Comment (Hmawn) [inline library/java/net/openid/appauth/TokenResponse.java:219]: *******

**Review by Hmawn — APPROVED** [review]

**Comment by Hmawn** [comment]
#130

**Pull request merged**